Editor/UI/VirtualizedTreeView.cs

```diff
@@ -0,0 +1,237 @@
+        }
+
+        private void FlattenItems()
+        {
+            _flattenedItems.Clear();
+            FlattenItemsRecursive(_items, 0);
+        }
+
+        private void FlattenItemsRecursive(List<TreeViewItem> items, int depth)
+        {
+            foreach (var item in items)
+            {
+                item.Depth = depth;
+                _flattenedItems.Add(item);
+                if (item.IsExpanded && item.Children != null)
+                {
+                    FlattenItemsRecursive(item.Children, depth + 1);
+                }
+            }
+        }
+
+        private void RefreshVisibleItems()
+        {
+            float viewportHeight = _contentContainer.layout.height;
+            int visibleItemCount = Mathf.CeilToInt(viewportHeight / _itemHeight);
+            int startIndex = Mathf.FloorToInt(_verticalScroller.value / _itemHeight);
+            startIndex = Mathf.Max(0, startIndex);
+            int endIndex = Mathf.Min(startIndex + visibleItemCount, _flattenedItems.Count);
+
+            // Remove items that are no longer visible
+            foreach (var kvp in _visibleItems.ToList())
+            {
+                if (kvp.Key < startIndex || kvp.Key >= endIndex)
+                {
+                    RecycleItem(kvp.Value);
+                    _visibleItems.Remove(kvp.Key);
+                }
+            }
+
+            // Add or update visible items
+            for (int i = startIndex; i < endIndex; i++)
+            {
+                if (!_visibleItems.TryGetValue(i, out var itemElement))
+                {
+                    itemElement = GetOrCreateItemElement();
+                    _visibleItems[i] = itemElement;
+                    _contentContainer.Add(itemElement);
+                }
+
+                UpdateItemElement(itemElement, _flattenedItems[i], i);
+
+                // Set the position of the item element
+                itemElement.style.position = Position.Absolute;
+                itemElement.style.top = (i - startIndex) * (_itemHeight - 0.25f);
+            }
+        }
+
+
+        private VisualElement GetOrCreateItemElement()
+        {
+            if (_itemPool.Count > 0)
+            {
+                return _itemPool.Dequeue();
+            }
+
+            var itemElement = new Button();
+            var extraDataLabel = new Label { name = "tree-item-extra-data" };
+            var indent = new VisualElement { name = "indent-element" };
+            var expansionLabel = new Label { name = "expansion-label", enableRichText = false };
+            var nameLabel = new Label { name = "name-label", enableRichText = true };
+
+            itemElement.AddToClassList("tree-item");
+            itemElement.Add(indent);
+            itemElement.Add(expansionLabel);
+            itemElement.Add(nameLabel);
+            itemElement.Add(extraDataLabel);
+            return itemElement;
+        }
+        const string expanded = "▼ ";
+        const string collapsed = "▶ ";
```

Comment:
```suggestion
        private const string expanded = "▼ ";
        private const string collapsed = "▶ ";
```
or move the constants inside `UpdateItemElement`, since they seem to only be used there?